pkg/api/vlabs/types.go

```diff
@@ -496,6 +498,16 @@ func (p *Properties) HasWindows() bool {
 	return false
 }
 
+// HasAvailabilityZones returns true if the cluster contains pools with zones
+func (p *Properties) HasAvailabilityZones() bool {
```

Comment:
another unit test opportunity :)

Reply:
This func could also potentially be generalized to us the func described above.